.github/workflows/common.yml

```diff
@@ -0,0 +1,37 @@
+name: common
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+env:
+  PACKAGEPATH: ${{ github.workspace }}/go/src/github.com/${{ github.repository }}
+  GOLANGCI_LINT_VERSION: v1.23.7
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [1.14.x]
+        platform: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.PACKAGEPATH }}
+      - name: Install Go
+        uses: actions/setup-go@v2-beta
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Install golangci-lint
+        # https://github.com/golangci/golangci-lint#binary
+        # golangci-lint binary will be installed at path ${{ github.workspace }}/bin with version ${{ env.GOLANGCI_LINT_VERSION }}
+        run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b ${{ github.workspace }}/bin ${{ env.GOLANGCI_LINT_VERSION }}
+      - name: Run golangci-lint
+        run: ${{ github.workspace }}/bin/golangci-lint run
+        working-directory: ${{ env.PACKAGEPATH }}
+      - name: Run unit tests
+        run: make unit
```

Comment:
do we also have to leverage `make compile` in one of the actions ?

Reply:
Nope, because by default, `go test` compiles and tests the package with source in the current directory, including tests, and runs the tests.

The package is built in a temporary directory so it does not interfere with the non-test installation.